src/Connector.AzureDataLake/Connector/AzureDataLakeConnector.cs

```diff
@@ -163,17 +455,18 @@ private void Flush(AzureDataLakeConnectorJobData configuration, string[] entityD
             var timestamp = DateTime.UtcNow.ToString("yyyy-MM-dd HH-mm-ss.fffffff");
             var fileName = $"{configuration.ContainerName}.{timestamp}.json";
 
-            _client.SaveData(configuration, content, fileName).GetAwaiter().GetResult();
+            _client.SaveData(configuration, content, fileName, JsonMimeType).GetAwaiter().GetResult();
         }
         
-        public override Task CreateContainer(ExecutionContext executionContext, Guid connectorProviderDefinitionId, IReadOnlyCreateContainerModelV2 model)
+        public override async Task CreateContainer(ExecutionContext executionContext, Guid connectorProviderDefinitionId, IReadOnlyCreateContainerModelV2 model)
         {
-            return Task.CompletedTask;
+            await Task.CompletedTask;
         }
 
         public override async Task ArchiveContainer(ExecutionContext executionContext, IReadOnlyStreamModel streamModel)
         {
             await _buffer.Flush();
+            await DeleteCacheTableIfExists(executionContext, streamModel);
```

Comment:
Do you know when archive is called? Do we need to keep the table, or flush a final parquet file to the lake?

Reply:
As per discsusion, it's caleld when you start the stream, which is counter intuitive